packages/opentelemetry-plugin-https/test/functionals/https-disable.test.ts

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NoopLogger } from '@opentelemetry/core';
+import { NodeTracer } from '@opentelemetry/node-sdk';
+import { Http } from '@opentelemetry/plugin-http';
+import * as assert from 'assert';
+import * as fs from 'fs';
+import * as https from 'https';
+import { AddressInfo } from 'net';
+import * as nock from 'nock';
+import * as sinon from 'sinon';
+import { plugin } from '../../src/https';
+import { DummyPropagation } from '../utils/DummyPropagation';
+import { httpsRequest } from '../utils/httpsRequest';
+
+describe('HttpPlugin', () => {
+  let server: https.Server;
+  let serverPort = 0;
+
+  describe('disable()', () => {
+    const httpTextFormat = new DummyPropagation();
+    const logger = new NoopLogger();
+    const tracer = new NodeTracer({
+      logger,
+      httpTextFormat,
+    });
+    before(() => {
+      nock.cleanAll();
+      nock.enableNetConnect();
+
+      plugin.enable((https as unknown) as Http, tracer, tracer.logger);
```

Comment:
because it's mysteriously not compatible with `Http` type that we specialize by doing (class HttpsPlugin extends `HttpPlugin` and class `HttpPlugin extends BasePlugin<Http>`). Even by doing something like `HttpPlugin<Https>`, Typescript is not happy because `Https` does not implement some methods that `Http` have.

Reply:
```
Argument of type 'typeof import("https")' is not assignable to parameter of type 'typeof import("http")'.
  Type 'typeof import("https")' is missing the following properties from type 'typeof import("http")': OutgoingMessage, ServerResponse, ClientRequest, IncomingMessage, and 3 more
```

Reply:
Hmm, I see. 

This is what I was thinking:

**Option 1:**
Include `Https` type to BasePlugin, something like this. `class HttpPlugin extends BasePlugin<Http | Https>` But we need to include `https` in `otel-plugin-http` package. I am not sure if we want to do this.

**Option 2:**

Modify `Plugin` interface somethign like this,
```
  enable<T>(
    moduleExports: T,
    tracer: Tracer,
    logger: Logger,
    config?: PluginConfig
  ): T;
```

Reply:
Another option is, what if we combine both the plugins (`http` and `https`) into one. Most of the codebase looks to be same except few wraps in the `https` plugin. I am not saying this is the best option, just wanted to put this comment for discussion.

Reply:
I think we should target to combine those plugins. It will be easier to maintain. We could start by having two and try two combines them in the in the meantime.

Reply:
Would you mind creating an issue to combine `http` and `https` plugins.

Reply:
#393